### PR TITLE
Use sbt-projectmatrix for cross-compilation

### PR DESCRIPTION
**Pull request by rtimush**
Currently `toguru-scala-client` has a hardcoded mapping between the Scala version and the Play Framework version. Ideally, we should have a core client that doesn't depend on Play at all, and then provide Play bindings for the supported scala and play version pairs.

This pull request prepares for this change by using `sbt-projectmatrix` plugin instead of the built-in sbt cross-compilation support. It will allow us to cross-build for different Play versions in the future but also has the advantage of being able to run compilation and tests for all scala versions in parallel.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,66 +1,65 @@
...
+      "-feature",
+      "-Xfatal-warnings",
+      "-Yno-adapted-args",
+      "-Xmax-classfile-name",
+      "130"
+    ),
+    libraryDependencies ++= Seq(
+      "org.scalaj"                 %% "scalaj-http"                 % "2.3.0",
+      "com.typesafe.scala-logging" %% "scala-logging"               % "3.5.0",
+      "commons-io"                 % "commons-io"                   % "2.4",
```
Comment (martinlechner1) [inline build.sbt]: While I see this: Are the dependencies reasonable?
Is this something for the future to reduce dependencies?

File: build.sbt [hunk 2]
```diff
@@ -1,66 +1,65 @@
...
+      "org.scalactic"              %% "scalactic"                   % "3.0.3",
+      "com.hootsuite"              %% "scala-circuit-breaker"       % "1.0.2",
+      "org.mockito"                % "mockito-core"                 % "2.0.8-beta" % "test",
+      "org.scalatest"              %% "scalatest"                   % "3.0.3" % "test",
+      "org.http4s"                 %% "http4s-dsl"                  % "0.17.0-M3" % "test",
+      "org.http4s"                 %% "http4s-blaze-server"         % "0.17.0-M3" % "test"
+    )
+  )
+  .jvmPlatform(
+    scalaVersions = Seq("2.12.3", "2.11.8"),
```
Comment (martinlechner1) [inline build.sbt]: Shall we go here to latest patch versions?

File: build.sbt (hunk 1 above)
Comment (rtimush) [reply build.sbt]: There are definitely too many dependencies, and some of them are not even used at all. That'll be the next step.

File: build.sbt (hunk 2 above)
Comment (rtimush) [reply build.sbt]: Absolutely, I just didn't want to squeeze different changes into a single PR.

**Pull request merged**